returnn/oggzip.py

```diff
@@ -1,104 +1,200 @@
-
-    @classmethod
-    def hash(cls, parsed_args):
-        del parsed_args["returnn_python_exe"]
-        return super().hash(parsed_args)
+__all__ = ["BlissToOggZipJob"]
+
+import os
+import shutil
+import subprocess as sp
+import tempfile
+import zipfile
+
+from i6_core.util import MultiOutputPath, relink
+
+from sisyphus import *
+
+Path = setup_path(__package__)
+
+
+class BlissToOggZipJob(Job):
+    """
+    This Job is a wrapper around the RETURNN tool bliss-to-ogg-zip.py.
+
+    """
+
+    __sis_hash_exclude__ = {"no_audio": False}
+
+    def __init__(
+        self,
+        bliss_corpus,
+        segments=None,
+        rasr_cache=None,
+        raw_sample_rate=None,
+        feat_sample_rate=None,
+        no_conversion=False,
+        no_audio=False,
+        returnn_python_exe=None,
+        returnn_root=None,
+    ):
+        """
+        use RETURNN to dump data into an ogg zip file
+
+        :param str|Path bliss_corpus: bliss corpus file
+        :param str|Path|MultiOutputPath segments: RASR segment file
+        :param str|Path rasr_cache: feature rasr cache
+        :param int raw_sample_rate: raw audio sampling rate
+        :param int feat_sample_rate: feature sampling rate
+        :param bool no_conversion: do not call the actual conversion, assume the audio files are already correct
+        :param bool no_audio: do not add audio files
+        :param Path|str returnn_python_exe: file path to the executable for running returnn (python binary or .sh)
+        :param Path|str returnn_root: file path to the RETURNN repository root folder
+        """
+        self.bliss_corpus = bliss_corpus
+        self.segments = segments
+        self.rasr_cache = rasr_cache
+        self.raw_sample_rate = raw_sample_rate
+        self.feat_sample_rate = feat_sample_rate
+        self.no_conversion = no_conversion
+        self.no_audio = no_audio
+        self.concurrent = (
+            len(segments.hidden_paths) if isinstance(segments, MultiOutputPath) else 1
+        )
+
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self.zip_subarchives = (
+            MultiOutputPath(
+                self,
+                "ogg.$(TASK).zip",
+                {i: "ogg.{}.zip".format(i) for i in range(1, self.concurrent + 1)},
+                cached=True,
+            )
+            if self.concurrent > 1
+            else None
+        )
+
+        self.out_ogg_zip = self.output_path("out.ogg.zip")
+
+        self.rqmt = None
+
+    def tasks(self):
+        if self.rqmt:
+            yield Task("run", rqmt=self.rqmt, args=range(1, self.concurrent + 1))
+        else:
+            yield Task("run", mini_task=True, args=range(1, self.concurrent + 1))

```

Comment:
these tasks could be heavy on I/O and disk space, even temporary. I would prefer these to be: `mini_task=False`

Reply:
ah sry only for the merge. for run there is already the choise

Reply:
`merge` is now not a mini task anymore by default.